base_classes/NXgoniometer.nxdl.xml

```diff
@@ -0,0 +1,74 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<?xml-stylesheet type="text/xsl" href="nxdlformat.xsl" ?>
+<!--
+# NeXus - Neutron and X-ray Common Data Format
+# 
+# Copyright (C) 2008-2025 NeXus International Advisory Committee (NIAC)
+# 
+# This library is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 3 of the License, or (at your option) any later version.
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this library; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+#
+# For further information, see http://www.nexusformat.org
+-->
+<definition xmlns="http://definition.nexusformat.org/nxdl/3.1" category="base"
+	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xsi:schemaLocation="http://definition.nexusformat.org/nxdl/3.1 ../nxdl.xsd"
+	name="NXgoniometer" 
+	type="group" extends="NXobject"
+	>
+
+	<doc>
+		Goniometer rotation angles and sample translations used during a
+		measurement. The goniometers used at neutron and x-ray
+		facilities are typically four-circle or six-circle goniometers,
+		or variants of either one. There is no universally recognized
+		naming convention for angles and translations, so the group
+		should contain a :ref:`NXcite` group, which describes how the
+		angles are defined.
+	</doc>
+
+    <group name="convention" type="NXcite">
+        <doc>
+            A reference to a publication or online documentation that
+            describe the convention used in this group to define the
+            names of angles and sample translations.
+        </doc>
+    </group>
+    <group type="NXcoordinate_system">
+        <doc>
+            The coordinate system used by the translation fields. This
+            group should be present if these fields do not use the
+            default NeXus Coordinate System.
+        </doc>
+    </group>
+    <field name="goniometer">
+        <doc>
+            The common name for the type of goniometer used in these
+            measurements.
+        </doc>
+    </field>
+	<field name="ROTATION_ANGLE" type="NX_FLOAT" units="NX_ANGLE" nameType="any">
+        <doc>
+            One of the goniometer rotation angles defined by the
+            specified convention.
+        </doc>
+    </field>
+    <field name="SAMPLE_TRANSLATION" type="NX_FLOAT" units="NX_LENGTH" nameType="any">
```

Comment:
Is this (vector?) field necessary? Can this be recorded by `NXtransformations` in `NXsample`?

Reply:
I have never used NXtransformations, so you may well be right that this is the way you should handle it if you do. My personal preference is that we encapsulate all the parameters that are required to transform the data from instrumental to reciprocal space coordinates in one place. This group contains all the information we need to construct the transformations described, e.g., [here](https://nexpy.github.io/nxrefine/introduction.html#experimental-geometry). I know that NXtransformations handle that process in other implementations, such as NXmx, but that is not how we do it.